lib/puma/client.rb

```diff
@@ -210,6 +215,17 @@ def try_to_parse_proxy_protocol
     end
 
     def try_to_finish
```

Comment:
This is a very important path so we'll need to be careful with anything we add here.